File: CustomViewportLayout/Visual Basic/CustomViewportLayout/MyModel.vb
Imports System.Collections.Generic
Imports System.Text
Imports System.Windows
Imports devDept.Eyeshot
Imports devDept.Graphics
Imports System.ComponentModel

Public Class MyModel
    Inherits Model
    Public Sub New()
    End Sub

    Protected Overrides Sub OnHandleCreated(e As EventArgs)
        InitializeViewportsByNumber(2)

        MyBase.OnHandleCreated(e)
    End Sub

    Public Overrides Sub UpdateViewportsSizeAndLocation()
        Dim height1 As Integer = CInt(2 * Size.Height / 3.0 - ViewportsGap / 2.0)
        Dim height2 As Integer = CInt(1 * Size.Height / 3.0 - ViewportsGap / 2.0)

        Viewports(0).Size = New System.Drawing.Size(CInt(Size.Width), height1)
        Viewports(1).Size = New System.Drawing.Size(CInt(Size.Width), height2)

        Viewports(1).Location = New System.Drawing.Point(0, height1 + ViewportsGap)
    End Sub

    Private Sub InitializeViewportsByNumber(numberOfViewports As Integer)
        If Viewports.Count > numberOfViewports Then
            While Viewports.Count > numberOfViewports
                Viewports.RemoveAt(Viewports.Count - 1)
            End While
        Else
            While Viewports.Count < numberOfViewports
                Viewports.Add(New Viewport())
            End While
        End If
    End Sub

End Class

File: CustomViewportLayout/CSharp/CustomViewportLayout/MyModel.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Windows;
using devDept.Eyeshot;
using devDept.Graphics;
using System.ComponentModel;

namespace WpfApplication1
{
    [ToolboxItem(true)/*, Designer(typeof(MyModelControlDesigner))*/]
    public class MyModel : Model
    {
        public MyModel()
        {
        }

        protected override void OnHandleCreated(EventArgs e)
        {
            InitializeViewportsByNumber(2);

            base.OnHandleCreated(e);
        }        

        public override void UpdateViewportsSizeAndLocation()
        {
            int height1 = (int)(2 * Size.Height / 3.0 - ViewportsGap / 2.0);
            int height2 = (int)(1 * Size.Height / 3.0 - ViewportsGap / 2.0);

            Viewports[0].Size = new System.Drawing.Size((int)Size.Width, height1);
            Viewports[1].Size = new System.Drawing.Size((int)Size.Width, height2);

            Viewports[1].Location = new System.Drawing.Point(0, height1 + ViewportsGap);
        }

        private void InitializeViewportsByNumber(int numberOfViewports)
        {
            if (Viewports.Count > numberOfViewports)
            {
                while (Viewports.Count > numberOfViewports)
                    Viewports.RemoveAt(Viewports.Count - 1);
            }
            else
            {
                while (Viewports.Count < numberOfViewports)
                    Viewports.Add(new Viewport());
            }
        }

    }
}